math/mathcore/src/BinData.cxx

```diff
@@ -267,44 +270,35 @@ namespace ROOT {
       }
       else
       {
+        // copy data vector and set correct pointer
         fData = rhs.fData;
-
         if ( !fData.empty() )
           fDataPtr = &fData.front();
 
+         // copy coordinate erro and set correct pointers
         fCoordErrors = rhs.fCoordErrors;
-        fDataError = rhs.fDataError;
-        fDataErrorHigh = rhs.fDataErrorHigh;
-        fDataErrorLow = rhs.fDataErrorLow;
-
-        if( ! fCoordErrors.empty() )
-        {
-          assert( kCoordError == fErrorType || kAsymError == fErrorType );
-          fCoordErrorsPtr.resize( fDim );
-
-          for ( unsigned int i=0; i<fDim; i++ )
-          {
-            fCoordErrorsPtr[i] = fCoordErrors[i].empty() ? nullptr : &fCoordErrors[i].front();
-          }
+        if (!fCoordErrors.empty()) {
+           assert(kCoordError == fErrorType || kAsymError == fErrorType);
+           fCoordErrorsPtr.resize(fDim);
+           for (unsigned int i = 0; i < fDim; i++) {
+              fCoordErrorsPtr[i] = fCoordErrors[i].empty() ? nullptr : &fCoordErrors[i].front();
+           }
         }
-
+        // copy data error
         fDataError = rhs.fDataError;
+        if (!fDataError.empty()) {
+           assert(kValueError == fErrorType || kCoordError == fErrorType);
+           fDataErrorPtr = &fDataError.front();
+        }
+        // copy the asymmetric data error
         fDataErrorHigh = rhs.fDataErrorHigh;
         fDataErrorLow = rhs.fDataErrorLow;
-
-        assert( fDataErrorLow.empty() == fDataErrorHigh.empty() );
-        assert( fDataErrorLow.empty() != fDataError.empty() || kNoError == fErrorType );
-
-        if ( !fDataError.empty() )
-        {
-          assert( kValueError == fErrorType || kCoordError == fErrorType );
-          fDataErrorPtr = &fDataError.front();
-        }
-        else if ( !fDataErrorHigh.empty() && !fDataErrorLow.empty() )
-        {
-          assert( kAsymError == fErrorType );
-          fDataErrorHighPtr = &fDataErrorHigh.front();
-          fDataErrorLowPtr = &fDataErrorLow.front();
+        // both error low and high should be empty or not
```

Comment:
Before we used to check that kNoError was equal to fErrorType when only one of fDataErrorLow and fDataErrorHigh was empty. Isn't this necessary anymore?